Refresher/Patching/Patcher.cs

```diff
@@ -123,74 +139,151 @@ private static List<PatchTargetInfo> FindUrl(Stream file)
             byte[] match = new byte[len];
             if (file.Read(match) < len) continue;
             string str = Encoding.UTF8.GetString(match).TrimEnd('\0');
-            
-            if(str.Contains('%')) continue; // Ignore printf strings, e.g. %s
 
-            Regex regex = UrlMatch();
-            MatchCollection matches = regex.Matches(str);
+            if (str.Contains('%')) continue; // Ignore printf strings, e.g. %s
 
-            if (matches.Count != 0)
-                found.Add(new PatchTargetInfo
+            if (UrlMatch().Matches(str).Count != 0)
+                foundUrls.Add(new PatchTargetInfo
                 {
                     Length = len,
                     Offset = foundPosition,
                     Data = str,
                 });
         }
 
-        return found;
+        List<PatchTargetInfo> foundDigests = new();
+
+        foreach (long foundPosition in foundPossibleCookiePositions)
+        {
+            file.Position = foundPosition;
+
+            byte[] cookieBuf = new byte[8];
+            //If we didnt read enough or what we read isnt "cookie\0\0"
+            if (reader.Read(cookieBuf) < cookieBuf.Length || !"cookie\0\0"u8.SequenceEqual(cookieBuf))
+            {
+                //Skip this possibility
+                continue;
+            }
+
+            const int checkSize = 1000;
+
+            //Go back half the check size in bytes (so that `cookie` is in the middle)
+            file.Position -= checkSize / 2;
+
+            byte[] checkArr = new byte[checkSize];
+            Span<byte> toCheck = checkArr.AsSpan().Slice(0, reader.Read(checkArr));
+
+            Regex regex = DigestMatch();
+            for (int i = 0; i < toCheck.Length; i++)
+            {
+                //Skip all null bytes
+                while (i < toCheck.Length && toCheck[i] == 0) i++;
+
+                int len = 0;
+                int start = i;
+
+                //Go over all non-null bytes
+                while (i < toCheck.Length && toCheck[i] != 0)
+                {
+                    len++;
+                    i++;
+                }
+
+                if (len != 18)
+                    continue;
+
+                string str = Encoding.UTF8.GetString(toCheck.Slice(start, len));
+
+                //If theres exactly 18 matches,
+                if (regex.Matches(str).Count == 18)
+                {
+                    //Then we have found a digest key
+                    foundDigests.Add(new PatchTargetInfo
+                    {
+                        Length = len,
+                        Offset = file.Position - checkSize + start,
+                        Data = str,
+                    });
+                }
+            }
+        }
+
+        return (foundUrls, foundDigests);
```

Comment:
Not a fan of having this digest code be a part of this function, we can definitely break this whole thing up into steps

Reply:
> Not a fan of having this digest code be a part of this function, we can definitely break this whole thing up into steps

Its mostly like that for speed, so we dont have to go through all the ELF sections twice to first look for URLs, then look for digest keys, IO is the bottleneck here and doubling our IO load would practically double patch time

Reply:
Oh, no, I don't mean to introduce any logic changes here, that can stay as-is. For example these iterations over `foundPossibleCookiePositions` and `foundPossibleUrlPositions`, those can be extracted into functions to help readability since this function is getting pretty long. Although linear, it's a bit hard to follow, if I'm honest.

Reply:
Split into multiple functions